allo/allo.py

```diff
@@ -1,3 +1,326 @@
+# -*- coding: utf-8 -*-
+"""
+This file serve as a starting point for a Python console script.
+"""
+import random
+
+from allo.room.room import Office, LivingSpace
+from allo.person.person import Fellow, Staff
+
+
 class Dojo(object):
-    pass
+    """
+    Class Dojo the main entry point to a110-cli
+    """
+    def __init__(self):
+        self.all_persons = {}
+        self.all_fellows = {}
+        self.all_staff = {}
+        self.all_rooms = {}
+        self.all_offices = {}
+        self.all_living_space = {}
+        self.room_allocation = {}
+        self.pending_room_allocation = {'office': {}, 'living_space': {}}
+
+    def create_room(self, room_type, room_name):
+        """
+        Creates a room when room type is specified to living space of office and the room name
+        :param room_type: string
+        :param room_name: string
+        :return: string
+        """
+        try:
+            room_type = room_type.lower()
+            room_name = room_name.lower()
+            if room_name in self.all_rooms.keys():
+                return "Sorry, {} room name has been taken.".format(room_name)
+            if room_type == 'office':
+                new_office = Office(room_name)
+                self.all_offices[room_name] = new_office
+                self.all_rooms[room_name] = new_office
+                return "{} - office added successfully".format(room_name)
+            elif room_type == 'living':
+                new_living = LivingSpace(room_name)
+                self.all_living_space[room_name] = new_living
+                self.all_rooms[room_name] = new_living
+                return "{} - living space added successfully".format(room_name)
+            else:
+                return "A room can only be office or living space"
+        except ValueError:
+            return "The value passed is not a string"
+
+    def rand_room_gen(self, room_type):
+        """
+        This takes in a room type and randomly chooses a room that is filled less to capacity
+        :param room_type: string
+        :return: room name if found else false
+        """
+        if room_type == 'office':
+            return self.rand_office_gen()
+        else:
+            return self.rand_living_space_gen()
+
+    def rand_living_space_gen(self):
```

Comment:
This can be part of `rand_room_gen`. Explore ways you can have less repetitive code.